### ISSUE TITLE
Check for duplicate parameter define calls

### ISSUE THREAD

**Issue by mnlevy1981** [marbl-ecosys/MARBL#108]
There is currently no check in `marbl_define_parameters()` to ensure that we are not defining the same parameter twice. We might want to update the `add_var()` routine to check that the variable being defined does not already exist. We should also consider the case-sensitivity of the parameters (see #107) -- should we be able to define `parm_fe_bioavail` if `parm_Fe_bioavail` is already defined?

Note that the stand-alone `put_get` regression test will capture cases where the same parameter is defined with two different names, but without an automated test suite we have no guarantee that developers will run the stand-alone tests.

Also, this applies to `marbl_define_config_vars()` as well. If the check goes into `add_vars()` then that should be picked up by `marbl_config_mod` automatically.

**Comment by mnlevy1981** [comment]
Note that the `associated()` function can be used to determine if two pointers point to the same variable:

```
if (associated(pa, pb)) then
   ! pa and pb point to same memory space
```

**Comment by mnlevy1981** [comment]
loop through previous defined variables, make sure `sname` is not previously defined and do the `associated(pa,pb)` check as well.

**Comment by mnlevy1981** [comment]
Closed via #181
